src/ext/resources/legacygraphqlapi/schema.graphqls

```diff
@@ -837,6 +873,23 @@ type Coordinates {
     lon: Float
 }
 
+input SearchWindowOptions @oneOf {
+    departure: SearchWindow
+    arrival: SearchWindow
+}
+
+input SearchWindow {
+    dateTime: DateTime
+    timeZone: TimeZone
+    before: SearchWindowRestriction
+    after: SearchWindowRestriction
+}
+
+input SearchWindowRestriction {
+    maxOffset: Duration
+    numberOfItineraries: Int
+}
```

Comment:
What should we do with paging if the search window and number of itineraries are defined like that. I guess the previous/next page logic is quite simple as we then search starting from the first/last included itinerary (depending on if it's arrive by search or not). However, deciding how many itineraries we should return for the next page is not as clear.

Also, does this model support the timetable search mode that @t2gran mentioned in https://github.com/opentripplanner/OpenTripPlanner/issues/4803#issuecomment-1429990094

Reply:
The paging would be ok, but I do not think the numOf before/after is a good idea. What the `numOf..` is when paging does not matter so much, adding the two numbers would work for the next/previous window. Note! The window is scaled to fit the `numOfIt.`. Is the use-case here to allow the UI implementer to include a few options before the time the user requests to show alternatives the user actually might reach? If that is the case I suggest:
```
input SearchWindow {
    // The instance of time the search ...
    time: TimeInstance
    // The initial length of the first search window
    length: Duration
    // The requested number of itineraries. The seach window wil be adjusted to fit the number requested.
    requestedSize : Int
    // Include a few itineraries before the time if the user have a change of bording it, this will be ignored 
    // when paging back/forward    
    grecePeriod : GracePeriodInput {
       length : Duration 
       maxSize : Int
    }
}
```

Reply:
Yes I think the use case is to allow returning some itineraries before the requested time. I think your suggestion might be clearer. However, I don't like the term size here as it's not really directly the size of the search window and I would expect the unit of size to be a duration.

Reply:
Agree, `numberOfItineraties` is much more clear - and keep in mind I am only trying to kick you to be sure you have thought of all options - in the end you decide what to do here.